specs/Menu.md

```diff
@@ -1,731 +1,545 @@
+  <div>custom item</div>
+</Menu>
+```
 
-Maybe should be named SubmenuContext?
-Should submenu have its own context?
+## Variants
 
-```TSX
-function() {
-return (
-    <OpenableContext.Provider value={submenuState}>
-        {props.children}
-    </OpenableContext.Provider>);
-}
-```
+### Nested menus
 
-#### SubmenuList
+A `Menu` should be able to trigger an additional instance of itself as a part of one or more of its options. The nested `Menu` component should have the same functional capabilities as the root `Menu` component.
 
-```TSX
-function() {
-    const openableContext = React.useContext(OpenableContext);
-    return (openableContext.open &&
-    <Slots.Root {...menuProps}>
-        {props.children}
-    </Slots.Root>);
-}
-```
+The actions that trigger the the nested `Menu` should be be consistent with the actions that can trigger any root `Menu` from a similar UI control.
 
-#### SubmenuItem
+We advise that no more than two nested `Menu` components be used, but this spec does not functionally apply that constrain to the implementation of the `Menu` component.
 
-```TSX
-function() {
-    const openableContext = React.useContext(OpenableContext);
-    return (
-    <Slots.Root {...menuItemProps} onClick={openableContext.toggleOpen}>
-        {props.children}
-    </Slots.Root>);
-}
+### Selection state
+
+A `Menu` should be able to track and represent the selection state of all or some of its options if required.
+
+When an options is associated with a selection state. The `Menu`, either root or nested, should control its dismiss behavior accordingly based on configuration.
+
+### Sections
+
+A `Menu` can be partitioned into sections using visible dividers in its list of options. Each section can contain a heading title that announces or briefly describes the options in the particular section
+
+### Secondary text
+
+An option of a `Menu` component should be able to declare additional secondary text that can provide additional context describing the option or its usage.
+
+For example a secondary text can be a label that shows a keyboard shortcut that will perform an equivalent action of the option of the `Menu` component.
+
+### Split option with nesting
+
+An option of a `Menu` component can trigger a nested `Menu` component and also perform its default action by splitting the option into two interactable areas that handle each action separately.
+
+### Disabled option(s)
+
+All options in a `Menu` component can be disabled and should provide a visible indication for this purpose. User interaction should be defined for disabled options
+
+### Scrollable
+
+A `Menu` should display a vertical scrollbar when the number of options exceeds the height of the component
+
+### Standalone/No surface
+
+A `Menu` can be used without the temporary popup surface and its trigger. This will allow `Menu` components to be permanent page content or used in custom surfaces with a wider range of UI components.
```

Comment:
I would vote for having this as a separate (sub)component - Menu.List?

Reply:
There is perhaps too many different layers of abstraction here. In the API section this is exactly why is being proposed. I'm not entirely sure we should map the requirements to actual components